srtcore/sync.h

```diff
@@ -625,18 +621,89 @@ class CGlobEvent
 public:
     /// Triggers the event and notifies waiting threads.
     /// Simply calls notify_one().
-    static void inline triggerEvent()
-    {
-        return g_Sync.notify_one();
-    }
+    static void triggerEvent();
 
     /// Waits for the event to be triggered with 10ms timeout.
     /// Simply calls wait_for().
-    static bool waitForEvent()
-    {
-        return g_Sync.lock_wait_for(milliseconds_from(10));
-    }
+    static bool waitForEvent();
+};
+
+////////////////////////////////////////////////////////////////////////////////
+//
+// CThread class
+//
+////////////////////////////////////////////////////////////////////////////////
+
+#ifdef USE_STDCXX_CHRONO
+typedef std::system_error CThreadException;
+using CThread = std::thread;
+#else // pthreads wrapper version
+typedef ::CUDTException CThreadException;
+
+class CThread
+{
+public:
+    CThread();
+    /// @throws std::system_error if the thread could not be started.
+    CThread(void *(*start_routine) (void *), void *arg);
+
+#if HAVE_FULL_CXX11
+    CThread& operator=(CThread &&other);
+#else
+    CThread& operator=(CThread &other);
+#endif
+
+public: // Observers
+    /// Checks if the CThread object identifies an active thread of execution.
+    /// A default constructed thread is not joinable.
+    /// A thread that has finished executing code, but has not yet been joined
+    /// is still considered an active thread of execution and is therefore joinable.
+    bool joinable() const;
+
+public:
+    /// Blocks the current thread until the thread identified by *this finishes its execution.
+    /// If that thread has already terminated, then join() returns immediately.
+    ///
+    /// @throws std::system_error if an error occurs
+    void join();
+
+private:
+    pthread_t m_thread;
+
 };
+#endif
+
+///
+/// @throws CThreadException if the thread could not be started.
+
+#ifdef USE_STDCXX_CHRONO
+//template< class Function, class... Args>
+//inline void CreateThread(CThread& th, Function&& f, Args&&... args)
```

Comment:
This function name looks like out-of-the-blue.

I think better use either a method of `CThread` that will assign a new thread to itself:
```
m_ThisThread.create(fn, args);
```
or just changed into a static member:
```
CThread::create(m_ThisThread, fn, args);
```

Reply:
It can't be `CThread` because I want to map CThread directly into `std::thread` in case of C++11. `std::thread` does not have this `create` member function.
sync.h lie 639:
```c++
#ifdef USE_STDCXX_CHRONO
typedef std::system_error CThreadException;
using CThread = std::thread;
```

Reply:
Renamed `CreateThread(..)` to `SetThread(..)` to avoid confusion with Windows API according to the internal discussion.